MigrationTutorial/Blog-Post.md

```diff
@@ -0,0 +1,658 @@
+
+</br>
+
+---
+
+#### *(11) V2 Migration function*
+```cs
+public static void DoMigrate(Migration migration, ulong oldSchemaVersion)
+{
+    var newEmployees = migration.NewRealm.All<Employee>(); // 14)
+    var oldEmployees = migration.OldRealm.DynamicApi.All("Employee");   // 15)
+
+    // 16) change string to enum
+    for (var i = 0; i < newEmployees.Count(); i++)
+    {
+        var newEmployee = newEmployees.ElementAt(i);
+        var oldEmployee = oldEmployees.ElementAt(i);
+        if (string.Equals(oldEmployee.Gender, "female", StringComparison.OrdinalIgnoreCase))
+        {
+            newEmployee.Gender = Gender.Female;
+        }
+        else if (string.Equals(oldEmployee.Gender, "male", StringComparison.OrdinalIgnoreCase))
+        {
+            newEmployee.Gender = Gender.Male;
+        }
+        else
+        {
+            newEmployee.Gender = Gender.Other;
+        }
+    }
+
+    var newConsumables = migration.NewRealm.All<Consumable>();
+    var oldConsumables = migration.OldRealm.DynamicApi.All("Consumable");
+    var distinctConsumableId = new HashSet<string>();
+    var consumableToDelete = new List<Consumable>();
+
+    // 17) rename Consumable.Price to Consumable.LastPurchasedPrice
+    migration.RenameProperty(nameof(Consumable), "Price", nameof(Consumable.LastPurchasedPrice));
+
+    // 18) remove duplicated Consumable to accommodate ProductId to become the new primary key
+    for (var i = 0; i < newConsumables.Count(); i++)
+    {
+        var currConsumable = newConsumables.ElementAt(i);
+
+        // remove duplicates since ProductId is the new PrimaryKey
+        if (distinctConsumableId.Contains(currConsumable.ProductId))
+        {
+            consumableToDelete.Add(currConsumable);
+        }
+        else
+        {
+            distinctConsumableId.Add(currConsumable.ProductId);
+        }
+    }
+
+    consumableToDelete.ForEach(x => migration.NewRealm.Remove(x));
+}
+```
+
+---
+
+</br>
+
+### V2 to V3
+
+#### V3 Models
+After some time, given that the company keeps expanding, a new `Department` called `Workshop` is introduced. This `Department` is now in charge of maintaining `MachineryAndTool`s [(12)](#12-model-v3machineryandtool). Because of this addition the company reclassifies two `Consumable`, `Brush` *19)* and `GlueHolder` *20)*, as `MachineryAndTool` since it is more environmentally friendly and economically advantageous to properly care for them. 
+
+</br>
+
+---
+
+#### *(12) Model V3.MachineryAndTool*
+```cs
+public class MachineryAndTool : RealmObject
+ {
+    [PrimaryKey]
+    public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+    private string _Type { get; set; }
+
+    public Type Type
```

Comment:
Would you prefer `ToolType`?
I don't think there's much value in showing the enum itself. Otherwise, `OperationalStatus` needs to be shown too, just bloating the blog-post. Do you disagree?